cloudbuild.yaml

```diff
@@ -0,0 +1,60 @@
+steps:
+  - id: 'fetch-cache'
+    name: gcr.io/cloud-builders/gcloud
+    entrypoint: sh
+    args:
+      - -c
+      - |
+        (
+          set -e
+          gsutil hash -h yarn.lock | grep md5 | tr -s " " | awk '{print $3}' > hashed.yarn-lock
+          gsutil -m cp "gs://hoprd-cache/$(cat hashed.yarn-lock)" cache.tar.gz 2> /dev/null
+          test -f cache.tar.gz
+          tar -zxf cache.tar.gz
+          echo "Using cache from: gs://hoprd-cache/$(cat hashed.yarn-lock)"
+        ) || true
+
+  - id: 'install-yarn'
+    name: node:16
+    waitFor: ['fetch-cache']
+    entrypoint: sh
+    args:
+      - -c
+      - |
+        test -f cache.tar.gz || yarn install
+
+  - id: 'build'
+    waitFor: ['install-yarn']
+    name: node:16
+    entrypoint: yarn
+    args: ['build']
+
+  - id: 'cache'
+    name: gcr.io/cloud-builders/gcloud
+    entrypoint: sh
+    args:
+      - -c
+      - |
+        (
+          set -e
+          test ! -f cache.tar.gz
+          tar -zcf cache.tar.gz node_modules
+          gsutil -m cp cache.tar.gz "gs://hoprd-cache/$(cat hashed.yarn-lock)"
```

Comment:
How do you prevent race conditions in terms of multiple runners writing that file at the same time?

Reply:
Good point, I don't. We probably need to include the PR name or similar as part of the `md5`, so it's written per PR rather than by simply `yarn.lock`